simd/src/Kokkos_SIMD.hpp

```diff
@@ -48,8 +48,13 @@
 #error "__AVX__ must be defined for KOKKOS_ARCH_AVX"
 #endif
 
+// FIXME_CRAY_LLVM - For some reason CrayClang does not correctly evaluate the
+// `__AVX2__` macro at this location even when the compiler supports it on the
+// given architecture. Avoiding this check for the (architecture, compiler)
+// combination for now. The code would fail if the macro is unavailable although
+// the error message might not be clean in the case.
```

Comment:
What backend is this for? Note that we are disabling all these checks for `OpenMPTarget` already.

Reply:
This is for enabling SIMD unit tests.

Reply:
Sure. Since you get here, you must try with a different backend than `OpenMPTarget`, right?

Reply:
I think this was reproduced even with just Serial backend.
I now wonder if this check against the cray compiler should be moved to L34 as well.

Reply:
It would make sense to me that the issue is not specific to `AVX2` but I'm surprised that AVX2 intrinsic are usable although the macro isn't defined.

Reply:
I don't know how to logically describe this behavior, but the macro was defined.
We checked that `__AVX2__` was in the list of preprocessor defined flags by dumping the output from CC.
Then we tested `#ifdef __AVX2__` right above this block, and it was correctly recognizing the macro to be defined; however, as soon as that added `ifdef` block was commented back out, the error on L58 was being outputted.

I had suspected that it could be something specific with the environment, but it sounded like @rgayatri23 was able to reproduce the same error on another machine with cray compiler.

Reply:
Ok an update - I was not able to reproduce it on a different machine, i.e., frontier which has the same CPU arch and same compiler/version. So I think it might be something on Perlmutter that is probably interfering with the `AVX2` macro evaluation.